scala-thoughts.md

```diff
@@ -0,0 +1,41 @@
+I found Scala to be very, very interesting.  It reminded me a lot of Haskell in 
+the way you can mix and match how you write operations, however it also really
+bothers me.  I understand how convenient and easy this makes it, but I think it
+can also lead to confusion.  Why is
+
+```
+(myList zip myOtherList).map
+```
+
+just as valid as
+
+```
+myList zip myOtherList map {}
+```
+
+?
+
+It seems to me that the additional flexibility comes at the cost of readability,
+and I suspect any large-scale business project that used this would have to 
+follow pretty strict conventions.  That being said, I think it wouldn't be too
+bad after I got used to it.
+
+Pattern matching was incredible.  If I had time I'd go back and do everything
+with pattern matching, except for the ones I wouldn't.  
+
+I found the `Option` type interesting as well - I remember liking that concept a
+lot in C# as well, and wish there was a better analogue in other strongly and 
+statically typed languages (if there is, please point me to it).  
```

Comment:
One downside of that is that you don't have to (although you should) explicitly check for `nullptr`, but you have to actually handle `Option` in order for it to compile